mmr/src/lib.rs

```diff
@@ -0,0 +1,211 @@
+use std::sync::LazyLock;
+
+use digest::{Digest, OutputSizeUser};
+use rpds::StackSync;
+
+const EMPTY_VALUE: [u8; 32] = [0; 32];
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct MMR<T, Hash, const MAX_HEIGHT: u8 = 32> {
+    roots: StackSync<Root>,
+    _hash: std::marker::PhantomData<(T, Hash)>,
+}
```

Comment:
The naming guidelines state that acronyms should only capitalize the first letter: https://rust-lang.github.io/api-guidelines/naming.html
e.g `MMR` should be `Mmr`. 

Spelling out the whole name is also not the worst thing: `MerkleMountainRange`

> In UpperCamelCase, acronyms and contractions of compound words count as one word: use Uuid rather than UUID, Usize rather than USize or Stdin rather than StdIn. In snake_case, acronyms and contractions are lower-cased: is_xid_start.